web/docs/auth/overview.md

```diff
@@ -504,8 +504,382 @@ try {
+    },
+  },
+}
+
+entity User {=psl
+  id Int @id @default(autoincrement())
+  username String @unique
+  password String
+  address String?
+psl=}
+```
+
+Then we'll export the `fields` object from the `server/auth/signup.ts` file:
+
+```ts title="server/auth/signup.ts"
+import { defineAdditionalSignupFields } from "@wasp/auth/index.js";
+
+export const fields = defineAdditionalSignupFields({
+  address: async (data) => {
+    const address = data.address;
+    if (typeof address !== "string") {
+      throw new Error("Address is required");
+    }
+    if (address.length < 5) {
+      throw new Error("Address must be at least 5 characters long");
+    }
+    return address;
+  },
+});
+```
+
+</TabItem>
+</Tabs>
+
+<small>
+
+Read more about the `fields` object in the [API Reference](#signup-fields-customization).
+</small>
+
+Keep in mind, that these field names need to exist on the `userEntity` you defined in your `main.wasp` file e.g. `address` needs to be a field on the `User` entity. 
+
+The field function will receive the data sent from the client and it needs to return the value of the field. If the field is invalid, the function should throw an error.
+
+:::info Using Validation Libraries
+
+You can use any validation library you want to validate the fields. For example, you can use `zod` like this:
+
+<details>
+<summary>Click to see the code</summary>
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js title="server/auth/signup.js"
+import { defineAdditionalSignupFields } from '@wasp/auth/index.js'
+import * as z from 'zod'
+
+export const fields = defineAdditionalSignupFields({
+  address: (data) => {
+    const AddressSchema = z
+      .string({
+        required_error: 'Address is required',
+        invalid_type_error: 'Address must be a string',
+      })
+      .min(10, 'Address must be at least 10 characters long');
+    const result = AddressSchema.safeParse(data.address);
+    if (result.success === false) {
+      throw new Error(result.error.issues[0].message);
+    }
+    return result.data;
+  },
+})
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="server/auth/signup.ts"
+import { defineAdditionalSignupFields } from '@wasp/auth/index.js'
+import * as z from 'zod'
+
+export const fields = defineAdditionalSignupFields({
+  address: (data) => {
+    const AddressSchema = z
+      .string({
+        required_error: 'Address is required',
+        invalid_type_error: 'Address must be a string',
+      })
+      .min(10, 'Address must be at least 10 characters long');
+    const result = AddressSchema.safeParse(data.address);
+    if (result.success === false) {
+      throw new Error(result.error.issues[0].message);
+    }
+    return result.data;
+  },
+})
+```
+</TabItem>
+</Tabs>
+</details>
+
+:::
+
+Now that we defined the fields, Wasp knows how to:
+
+1. Validate the data sent from the client
+2. Save the data to the database
```

Comment:
Nice, this is clear. I would periods at the end of sentences though. Remember, if sentence starts with capital letter, it should end with period (or something similar).

Reply:
Adjusted this piece as well 👍